lighthouse-core/test/gather/driver-test.js

```diff
@@ -40,20 +40,28 @@ function makePromiseInspectable(promise) {
     throw err;
   });
 
-  inspectablePromise.isDone = () => isResolved || isRejected;
-  inspectablePromise.isResolved = () => isResolved;
-  inspectablePromise.isRejected = () => isRejected;
-  inspectablePromise.getDebugValues = () => ({resolvedValue, rejectionError});
-
-  return inspectablePromise;
+  return Object.assign(inspectablePromise, {
```

Comment:
This pattern allows type script to infer the type correctly, while not changing the prototype of the return value (it is still a Promise).